pkg/fanal/image/remote.go

```diff
@@ -67,31 +67,40 @@ func tryRemote(ctx context.Context, imageName string, ref name.Reference, option
 
 }
 
-func parsePlatform(ref name.Reference, p string) (*v1.Platform, error) {
+func parsePlatform(ref name.Reference, options []remote.Option, img v1.Image, p string) (*v1.Platform, error) {
 	// OS wildcard, implicitly pick up the first os found in the image list.
 	// e.g. */amd64
 	if strings.HasPrefix(p, "*/") {
-		index, err := remote.Index(ref)
+		index, err := remote.Index(ref, options...)
 		if err != nil {
-			// Not a multi-arch image
-			if _, ok := err.(*remote.ErrSchema1); ok {
-				log.Logger.Debug("Ignored --platform as the image is not multi-arch")
-				return nil, nil
+			// Image is not a multi-arch image, but we can extract the OS from the image's config file.
+			cfg, err := img.ConfigFile()
```

Comment:
Why do we need to detect OS for single platform? We can just ignore --platform in the same way as Docker.

Reply:
Yes... we can ignore the platform for single platform image, but as long as the Trivy continues to scan with out fail with only available one single platform image, we are good @knqyf263

Reply:
Fixed in https://github.com/aquasecurity/trivy/pull/3326/commits/ade48b96a6ae83064386c8789abf7b2395333f7d

Reply:
@knqyf263, added the skip for not adding platform into remoteOpts if parsePlatform is nil under tryRemote

```
//Ignored --platform should skip adding platform remoteOpts
		if s != nil {
			remoteOpts = append(remoteOpts, remote.WithPlatform(*s))
		}
```